### PR TITLE
Added files for homework 2

### PR DESCRIPTION
**Pull request by iboreysha**
Added files for homework 2

### CODE REVIEW

File: homework-2/index.js [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+  res.send(users);
+
+});
+
+
+//
+// Get one user by ID
+//
+app.get('/api/users/:id', (req, res) => {
```
Comment (Ivanko5417) [inline homework-2/index.js:67]: you can use router from express for combining all routes for users

File: homework-2/index.js [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+  user.isDeleted = req.body.isDeleted
+
+  res.send(user);
+
+});
+
+
+//
+// Delete user completely
+//
```
Comment (Ivanko5417) [inline homework-2/index.js:135]: we should use soft delete, insted of hard one

File: homework-2/index.js [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+
+});
+
+
+//
+// Update user
+//
+app.put('/api/users/:id', (req, res) => {
+
+  const user = users.find(c => c.id === parseInt(req.params.id));
```
Comment (Ivanko5417) [inline homework-2/index.js:111]: please, filter by isDeleted field for all endpoints

File: homework-2/index.js [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+    res.status(400).send(error.details[0].message);
+    return;
+  }
+
+  user.login = req.body.login,
+  user.password = req.body.password,
+  user.age = req.body.age,
+  user.isDeleted = req.body.isDeleted
+
+  res.send(user);
```
Comment (Ivanko5417) [inline homework-2/index.js:128]: please, do not return `isDeleted` filed for all endpoints that return user

**Review by Ivanko5417 — CHANGES_REQUESTED** [review]

File: homework-2/index.js (hunk 2 above)
Comment (iboreysha) [reply homework-2/index.js]: Remove block  "Delete user completely"

File: homework-2/index.js [hunk 5]
```diff
@@ -56,17 +57,27 @@ app.get('/', (req, res) => {
...
 
-  res.send(users);
+  // Filtered non deleted users
+  const usersList = users.filter((user) => user.isDeleted === "false");
+  res.send(usersList);
+
+  // For testing - See all users: active and deleted 
+  console.log("----------------------");
+  console.log(users);
+  console.log("----------------------");
```
Comment (Ivanko5417) [inline homework-2/index.js:67]: let's remove unnecessary logs

File: homework-2/index.js [hunk 6]
```diff
@@ -56,17 +57,27 @@ app.get('/', (req, res) => {
 //
 app.get('/api/users', (req, res) => {
 
-  res.send(users);
+  // Filtered non deleted users
+  const usersList = users.filter((user) => user.isDeleted === "false");
```
Comment (Ivanko5417) [inline homework-2/index.js:61]: isDeleted must be boolean

File: homework-2/index.js [hunk 7]
```diff
@@ -56,17 +57,27 @@ app.get('/', (req, res) => {
...
 
 //
-// Get one user by ID
+// Get user by ID
 //
 app.get('/api/users/:id', (req, res) => {
 
-  const user = users.find((user) => user.id === req.params.id);
+  // Filtered non deleted users
+  const usersList = users.filter((user) => user.isDeleted === "false");
```
Comment (Ivanko5417) [inline homework-2/index.js:78]: you can also move this logic to separate function

**Review by Ivanko5417 — CHANGES_REQUESTED** [review]

File: homework-2/index.js [hunk 8]
```diff
@@ -0,0 +1,190 @@
...
+
+//
+// Validation schema
+//
+function validateUser(user) {
+  const schema = Joi.object({
+    login: Joi.string().min(3).max(20).required(),
+    password: Joi.string().min(3).max(20).alphanum().required(),
+    age: Joi.number().integer().min(4).max(130).required()
+    //isDeleted: Joi.boolean().required()
```
Comment (Ivanko5417) [inline homework-2/index.js:179]: i think it should be uncommented

**Review by Ivanko5417 — CHANGES_REQUESTED** [review]

**Review by Ivanko5417 — APPROVED** [review]

**Pull request merged**